db/blob/blob_file_reader.cc

```diff
@@ -220,14 +228,16 @@ Status BlobFileReader::ReadFooter(uint64_t file_size,
 
 Status BlobFileReader::ReadFromFile(const RandomAccessFileReader* file_reader,
                                     uint64_t read_offset, size_t read_size,
-                                    Slice* slice, Buffer* buf,
-                                    AlignedBuf* aligned_buf) {
+                                    Statistics* statistics, Slice* slice,
+                                    Buffer* buf, AlignedBuf* aligned_buf) {
   assert(slice);
   assert(buf);
   assert(aligned_buf);
 
   assert(file_reader);
 
+  RecordTick(statistics, BLOB_DB_BLOB_FILE_BYTES_READ, read_size);
```

Comment:
The actual number of bytes read depends on the result of the `file_reader->Read()` call below, right?

Reply:
Right. I feel the semantics of some of our counters are not really well-defined, i.e. is it supposed to be the number of bytes we want to read/write or is it the number of bytes we end up reading/writing? I guess we can get away with such ambiguity because any discrepancy between the two would be a result of an error (which should be rare)...